src/core/meshcop/tcat_agent.cpp

```diff
@@ -436,7 +445,21 @@ Error TcatAgent::HandleSingleTlv(const Message &aIncomingMessage, Message &aOutg
         case kTlvGetProvisioningURL:
             error = HandleGetProvisioningUrl(aOutgoingMessage, response);
             break;
-
+        case kTlvPresentPskdHash:
+            error = HandlePresentPskdHash(aIncomingMessage, offset, length);
```

Comment:
Can consider using `OffsetRange` (which encapsulates an offset along with a length) instead of passing `offset, length` as separate input arguments to each method?

There are a bunch of helper methods (e.g. `Message::Read<>` methods that work with `OffsetRange` directly).

Reply:
@abtink I'm not saying no (always open to suggestions) but it seems that this function: https://github.com/openthread/openthread/blob/main/src/core/common/message.cpp#L667 
behaves differently than this:
https://github.com/openthread/openthread/blob/main/src/core/common/message.cpp#L662

The second one always populates the buffer and returns an error if the requested length exceeds the buffer's content. First however doesn't populate the buffer in such case and returns error. Was it intended?

Reply:
Thanks, @canisLupus1313.

Regarding the use of `OffsetRange`, I just wanted to let you about this. I noticed that `offset` and `length` are already used as separate parameters in existing `TcatAgent` methods. So may make sense to keep it the same for now and then can consider to maybe change it in a future PR.

About the `Message::Read()`, you're right about the difference in behavior. We should update the documentation for `Read(OffsetRange &)` to clarify that it requires the full length bytes to be available for reading.

In general, I think for reading partial data `Message::ReadByte()` would be preferable since it returns the number of bytes read. The `Error Message::Read(...)` methods only return an `Error`,  making it impossible for the caller to know how many bytes were actually read.  Ideally, callers shouldn't really rely on the behavior of these methods when an error occurs (whether partial bytes are read or not). I may update this in an upcoming PR to clarify this in the documenation. Do you happen to know of any place `Message::Read()` is used (e.g. in `Tcat`) for partial read?

Reply:
I have to create a ticket in openthread space to track all technical debt created here to pay it later.
`HandleTransport` transport in `ble_secure.cpp` uses `Read` to get partial bytes. I don't recall any other.